runtime/ints.c

```diff
@@ -136,7 +141,20 @@ CAMLprim value caml_int_compare(value v1, value v2)
 
 CAMLprim value caml_int_of_string(value s)
 {
-    return Val_long(parse_intnat(s, 8 * sizeof(value) - 1, INT_ERRMSG));
+  intnat res;
+  if (0 == parse_intnat(s, 8 * sizeof(value) - 1, &res)) {
+    return Val_long(res);
+  }
+  caml_failwith(INT_ERRMSG);
```

Comment:
Style comment: I would test for `0 !=` and reorder the cases in these functions. It is more common and, I think, easier to read to have early returns or exits for the error cases, and let the successful code path reach the end of the function.

Reply:
This is a good idea. Done.